mne/source_estimate.py

```diff
@@ -3347,7 +3349,7 @@ def stc_near_sensors(evoked, trans, subject, distance=0.01, mode='sum',
     subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
     if src is None:  # fake a full surface one
         rrs = [read_surface(op.join(subjects_dir, subject,
-                                    'surf', f'{hemi}.pial'))[0]
+                                    'surf', f'{hemi}.{surface}'))[0]
```

Comment:
This isn't quite the proposed solution [here](https://github.com/mne-tools/mne-python/issues/9971#issuecomment-962644529). It would be more like:
```
if surf is not None:
    surf_rrs = [...]
    rrs = surf_rrs
if src is None:
    _validate_type(surf, str, 'surf', extra='when src is None')
    rrs = surf_rrs
    ...
else:
    if surf is None:
        rrs = np.concatenate([s['rr'][s['vertno']] for s in src])
```
if surf is not None:
    rrs = ...